File: Eulerpy/Euler7.py
#!/usr/bin/python

import math

def primes_sieve(limit):
    limitn = limit+1
    not_prime = set()
    primes = []

    for i in range(2, limitn):
        if i in not_prime:
            continue

        for f in range(i*2, limitn, i):
            not_prime.add(f)

        primes.append(i)

    return primes




j = 1
while( len(primes_sieve(j)) < 10000):
	j = j + 1



print primes_sieve(j)

File: Eulerjl/Euler7.jl

function primes_sieve(limit)
	not_prime = Set()
	primes = []
	for i in 2:limit
		if (i in not_prime)
			continue
		end
		f = 2i
		while(f<limit)
			push!(not_prime,f)
			f += i
		end
		push!(primes, i)
	end
	return primes
end

j = 10000000^5
while(length(primes_sieve(j)) < 10000)
	j += 1
end
println(primes_sieve(j))

